target/linux/qualcommax/ipq807x/base-files/etc/hotplug.d/ieee80211/11_fix_wifi_mac

```diff
@@ -23,4 +23,9 @@ case "$board" in
 		[ "$PHYNBR" = "0" ] && macaddr_add $(get_mac_label) 2 > /sys${DEVPATH}/macaddress
 		[ "$PHYNBR" = "1" ] && macaddr_add $(get_mac_label) 3 > /sys${DEVPATH}/macaddress
 		;;
+	tplink,deco-x80-5g)
+		base_mac=$(get_mac_binary /tmp/factory_data/default-mac 0)
+		macaddr_add $base_mac $(expr "$PHYNBR" + 1) > /sys${DEVPATH}/macaddress
```

Comment:
I would prefer moving this to caldata, you can look at examples there how to patch the MAC so we dont need this fixup.

Reply:
I tried to patch that previously, but I don't think the partition is mounted in time to patch caldata using the hotplug.d script.

I can try again

Reply:
It has to be mounted since you are mounting it in preinit, otherwise caldata loading would have failed as well

Reply:
I found the partition mounting to pull the mac binary unreliable it did not work every time.
It seems ubiatatch exits with an exitcode but the partition is not ready by the time the mound command operates, and it fails to find the MTD device. some of the time.

I added a sleep 1 in the script and it fixes the problem but the get mac binary does not work for the lan as that inits well before the partition is up and it has already got a random mac by that time.

I'm not sure how to fix the lan mac apart from writing it into bootargs in and modifying the dts.

Reply:
Hm, can you try defining a fixed partition table and then using `compatible = "linux,ubi";` as that will make the kernel automount that UBI volume and that should happen really early as well.

Reply:
ok, do I have to define the whole partition map or just the partitions I want to mount as this has an smem table?

This is the boot mountingit is after the fallback mac:
`
[    4.417036] GMAC6(ffffff800400b900) Invalid MAC@ - using fe:c0:c3:3c:71:b4
[    4.626315] init: - preinit -
[    5.060429] ubi1: attaching mtd15
[    5.110661] ubi1: scanning is finished
[    5.117811] ubi1: attached mtd15 (name "factory_data", size 9 MiB)
[    5.117852] ubi1: PEB size: 131072 bytes (128 KiB), LEB size: 126976 bytes
[    5.122891] ubi1: min./max. I/O unit sizes: 2048/2048, sub-page size 2048
[    5.129781] ubi1: VID header offset: 2048 (aligned 2048), data offset: 4096
[    5.136618] ubi1: good PEBs: 72, bad PEBs: 0, corrupted PEBs: 0
[    5.143376] ubi1: user volume: 1, internal volumes: 1, max. volumes count: 128
[    5.149284] ubi1: max/mean erase counter: 16/9, WL threshold: 4096, image sequence number: 1097309032
[    5.156578] ubi1: available PEBs: 0, total reserved PEBs: 72, PEBs reserved for bad PEB handling: 20
[    5.165883] ubi1: background thread "ubi_bgt1d" started, PID 760
[    5.429679] UBIFS (ubi1:0): recovery needed
[    5.477984] UBIFS (ubi1:0): recovery deferred
[    5.478100] UBIFS (ubi1:0): UBIFS: mounted UBI device 1, volume 0, name "ubi_factory_data", R/O mode
[    5.481324] UBIFS (ubi1:0): LEB size: 126976 bytes (124 KiB), min./max. I/O unit sizes: 2048 bytes/2048 bytes
[    5.490548] UBIFS (ubi1:0): FS size: 4825088 bytes (4 MiB, 38 LEBs), max 48 LEBs, journal size 1142785 bytes (1 MiB, 9 LEBs)
[    5.500351] UBIFS (ubi1:0): reserved for root: 0 bytes (0 KiB)
[    5.511618] UBIFS (ubi1:0): media format: w4/r0 (latest is w5/r0), UUID 555FE482-AC27-41A3-A0C3-47CC50ED88FA, small LPT model
`

Reply:
fixed partitions still did not fix the network mac issue but now I get the error ` OF: Bad cell count for /soc@0/nand-controller@79b0000/nand@0/partitions` during the boot log but seemingly everything still works?
is this something to do with bad blocks or do i have a bad partition map or something ?

Reply:
found error with me baking in config during testing causing wan/ lan mac addresses to not be updated at boot.